protocol/cors/preflight.feature

```diff
@@ -0,0 +1,38 @@
+Feature: Server must support HTTP OPTIONS for CORS preflight requests
+
+  Background: Set up test container
+    * def testContainer = rootTestContainer.createContainer()
+
+  Scenario: OPTIONS request returns headers for pre-flight check
+    Given url testContainer.url
+    And header Origin = 'https://tester'
+    And header Access-Control-Request-Method = 'POST'
+    And header Access-Control-Request-Headers = 'X-CUSTOM, Content-Type'
+    When method OPTIONS
+    Then match [200, 204] contains responseStatus
+    And match header Access-Control-Allow-Origin == 'https://tester'
+    And match header Access-Control-Allow-Methods contains 'POST'
+    And match header Access-Control-Allow-Headers contains 'X-CUSTOM'
+    And match header Access-Control-Allow-Headers contains 'Content-Type'
+    And match header Access-Control-Allow-Credentials == 'true'
+    # We should check the list of headers exposed but what is the required list
+    And match header Access-Control-Expose-Headers != null
+    And match response == ''
+
+  # See https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS - "CORS-preflight requests must never include credentials." - what should happen then?
```

Comment:
If request includes credentials, server shouldn't include `Access-Control-Allow-Credentials` in the response. I think...

Reply:
https://fetch.spec.whatwg.org/#http-responses seems to suggest that `Access-Control-Allow-Credentials` should be included in the response to preflight requests and servers appear to be doing that. I just wasn't sure if sending unnecessary credentials in the request was an error or simply redundant.

Reply:
The table with examples in https://fetch.spec.whatwg.org/#cors-protocol-and-credentials gives a bit more detail. There are cases that the header may be omitted - if I'm reading that correctly.